### ISSUE TITLE
Delete does not work in 'Spell Check'

### ISSUE THREAD

**Issue by Josef-Meier** [Program-O/Program-O#426]
on version: 2.6.8
Delete not working in "Spell Check"and wrong mouseover text for delete-icon (Action) in Spell Check:

Possible solution
Change following in spellcheck.php after linenumber 301:
    1) change action=del in action=delete
    2) change text alt=Edit... in alt= Delete...
    3) change text title=Edit... in title=Delete...
    
    ----

    foreach ($result as $row)
    {
        $i++;
        $misspell = _strtoupper($row['missspelling']);
        $correction = _strtoupper($row['correction']);
        $id = $row['id'];
        $group = round(($id / 50));

        //// J. Meier changing because delete not works and wrong mouseover text
        // $action = "<a href=\"index.php?page=spellcheck&amp;action=edit&amp;id=$id&amp;group=$group#$id\"><img src=\"images/edit.png\" border=0 width=\"15\" height=\"15\" alt=\"Edit this entry\" title=\"Edit this entry\" /></a>
        //            <a href=\"index.php?page=spellcheck&amp;action=del&amp;id=$id&amp;group=$group#$id\" onclick=\"return confirm('Do you really want to delete this missspelling? You will not be able to undo this!')\";><img src=\"images/del.png\" border=0 width=\"15\" height=\"15\" alt=\"Edit this entry\" title=\"Edit this entry\" /></a>";
        $action = "<a href=\"index.php?page=spellcheck&amp;action=edit&amp;id=$id&amp;group=$group#$id\"><img src=\"images/edit.png\" border=0 width=\"15\" height=\"15\" alt=\"Edit this entry\" title=\"Edit this entry\" /></a>
                   <a href=\"index.php?page=spellcheck&amp;action=delete&amp;id=$id&amp;group=$group#$id\" onclick=\"return confirm('Do you really want to delete this missspelling? You will not be able to undo this!')\";><img src=\"images/del.png\" border=0 width=\"15\" height=\"15\" alt=\"Delete this entry\" title=\"Delete this entry\" /></a>";
        
        $htmltbl .= "<tr valign=top>
                            <td>$misspell</td>
                            <td>$correction</td>
                            <td align=center>$action</td>
                        </tr>";
    }

**Comment by patrickschur** [comment]
Thanks, great work. I will fix this issue. Feel free to make pull requests if you want. :+1:

**Comment by Josef-Meier** [comment]
@patrickschur
Version 2.6.10  
Hello,
In the last patched "spellcheck.php" there is a problem.
"Add a Misspelled word" does not work.
I have identified the problem in line 385.
It seems it need a NULL value as in the previous version before the last patch.
If I change the line, then "spellcheck" works for me.
I have changed line 385 into:
$sql = "INSERT INTO spellcheck VALUES (NULL, :missspell, :correction)";
Possibly, the issue # 426 needs to be reopened.
Best Regards
